zk/stages/stage_sequence_execute.go

```diff
@@ -25,6 +26,72 @@ func SpawnSequencingStage(
 	cfg SequenceBlockCfg,
 	historyCfg stagedsync.HistoryCfg,
 	quiet bool,
+) (err error) {
+	roTx, err := cfg.db.BeginRo(ctx)
+	if err != nil {
+		return err
+	}
+	defer roTx.Rollback()
+
+	lastBatch, err := stages.GetStageProgress(roTx, stages.HighestSeenBatchNumber)
+	if err != nil {
+		return err
+	}
+
+	highestBatchInDS, err := cfg.datastreamServer.GetHighestBatchNumber()
+	if err != nil {
+		return err
+	}
+
+	if !cfg.zk.SequencerResequence || lastBatch >= highestBatchInDS {
```

Comment:
I think we run the risk here of the sequencer just continuing on after resequencing is complete with this logic.  Ideally we want a hard stop once resequencing is complete and the node will need restarting without the resequence flag being set.

Reply:
Thanks, this is a great point! I've added a check at the beginning and the node will stop sequencing new blocks after re-sequencing completes.